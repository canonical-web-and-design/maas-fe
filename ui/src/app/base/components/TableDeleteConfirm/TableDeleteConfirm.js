import PropTypes from "prop-types";
import React from "react";

import Button from "app/base/components/Button";
import Col from "app/base/components/Col";
import Row from "app/base/components/Row";

const TableDeleteConfirm = ({ modelName, modelType, onCancel, onConfirm }) => {
  return (
    <Row>
      <Col size="7">
        Are you sure you want to delete {modelType} "{modelName}"?{" "}
        <span className="u-text--light">
          This action is permanent and can not be undone.
        </span>
      </Col>
      <Col size="3" className="u-align--right">
        <Button onClick={onCancel}>Cancel</Button>
        <Button appearance="negative" onClick={onConfirm}>
          Delete
        </Button>
      </Col>
    </Row>
  );
};

TableDeleteConfirm.propTypes = {
  modelName: PropTypes.string,
  modelType: PropTypes.string,
  onCancel: PropTypes.func,
  onConfirm: PropTypes.func
};

export default TableDeleteConfirm;
