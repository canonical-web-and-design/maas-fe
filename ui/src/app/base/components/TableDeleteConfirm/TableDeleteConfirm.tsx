import { ActionButton, Button, Col, Row } from "@canonical/react-components";

import { COL_SIZES } from "app/base/constants";
import { useCycled } from "app/base/hooks";

type Props = {
  deleted: boolean;
  deleting: boolean;
  message?: string;
  modelName?: string;
  modelType?: string;
  onClose: () => void;
  onConfirm: () => void;
  sidebar?: boolean;
};

const TableDeleteConfirm = ({
  deleted,
  deleting,
  message,
  modelName,
  modelType,
  onClose,
  onConfirm,
  sidebar = true,
}: Props): JSX.Element => {
  useCycled(deleted, () => {
    onClose();
  });
  const { DELETE_ROW_BUTTONS, SIDEBAR, TOTAL } = COL_SIZES;
  return (
    <Row>
      <Col
        size={
          sidebar
            ? TOTAL - SIDEBAR - DELETE_ROW_BUTTONS
            : TOTAL - DELETE_ROW_BUTTONS
        }
      >
        <p className="u-no-margin--bottom u-no-max-width">
          {message ||
            `Are you sure you want to delete ${modelType} "${modelName}"?`}{" "}
          <span className="u-text--light">
            This action is permanent and can not be undone.
          </span>
        </p>
      </Col>
      <Col size={DELETE_ROW_BUTTONS} className="u-align--right">
        <Button className="u-no-margin--bottom" onClick={onClose}>
          Cancel
        </Button>
        <ActionButton
          appearance="negative"
          className="u-no-margin--bottom"
          data-test="delete-confirm"
          loading={deleting}
          onClick={onConfirm}
          success={deleted}
        >
          Delete
        </ActionButton>
      </Col>
    </Row>
  );
};

export default TableDeleteConfirm;
