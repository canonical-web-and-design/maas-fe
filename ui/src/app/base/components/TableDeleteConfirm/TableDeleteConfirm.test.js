import { shallow } from "enzyme";
import React from "react";

import TableDeleteConfirm from "./TableDeleteConfirm";

describe("TableDeleteConfirm", () => {
  it("renders", () => {
    const wrapper = shallow(
      <TableDeleteConfirm
        modelName="Cobba"
        modelType="user"
        onCancel={jest.fn()}
        onConfirm={jest.fn()}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
