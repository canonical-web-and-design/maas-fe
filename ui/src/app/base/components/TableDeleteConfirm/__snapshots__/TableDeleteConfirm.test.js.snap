// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableDeleteConfirm renders 1`] = `
<Row>
  <Col
    size="7"
  >
    Are you sure you want to delete 
    user
     "
    Cobba
    "?
     
    <span
      className="u-text--light"
    >
      This action is permanent and can not be undone.
    </span>
  </Col>
  <Col
    className="u-align--right"
    size="3"
  >
    <Button
      onClick={[MockFunction]}
    >
      Cancel
    </Button>
    <Button
      appearance="negative"
      onClick={[MockFunction]}
    >
      Delete
    </Button>
  </Col>
</Row>
`;
